File: sdk/node/src/echo.ts
import { Agent } from "./agent.js";

const token = process.env.AGENT_TOKEN;
if (!token) throw new Error("Set AGENT_TOKEN to the one-time token shown by aTalk");

const baseUrl = process.env.ATALK_BASE_URL;
const agent = new Agent({ token, ...(baseUrl ? { baseUrl } : {}) });
agent.on("message", async (message) => {
  await message.reply("Hello human!");
});
agent.on("error", (error) => console.error(error.message));

await agent.start();
console.log("aTalk echo agent connected");

File: sdk/python/examples/echo.py
import os

from atalk import Agent

agent = Agent(token=os.environ["AGENT_TOKEN"], base_url=os.getenv("ATALK_BASE_URL", "http://127.0.0.1:4001"))


@agent.on_message
async def handle(message):
    if message.is_supervisor:
        await message.relay(message.text)
        return
    await message.reply("Hello human!")


@agent.on_error
async def handle_error(error):
    print(f"aTalk runtime error: {error}")


agent.run()
